feat: PCR primers

ivan-aksamentov opened the pull request:
Resolves: #119 

This PR adds PCR primers into consideration.
In particular it:

 - [x] converts original data to JSON format
 - [ ] ~highlights primer ranges in sequence views~
 - [x] detects and reports mutations in primers
 - [x] excludes primer changes when primer contains matching ambiguous nucleotide
 - [x] formats primer changes in export files
 - [x] shows primer changes as the tree node attribute and as a coloring (presence yes/no)

rneher commented:
This looks great, @ivan-aksamentov!

The one question that remains for me is whether the ambiguous non-ACGT states are handled correctly on reverse complementation. Their position needs to be mirrored and the state changed. As far as I can tell, this would need to happen here:

https://github.com/nextstrain/nextclade/pull/198/files#diff-00979bfe639fb449fb484e300313a75eR120

state 'R' for example could be 'A' and 'G', after complementation it is 'T', 'C' or state 'Y'. Similarly, the position 3 in a reverse primers would be position -3 from the end after complementation.

ivan-aksamentov commented:
Addressed the above issues:

 - [x] ambiguous nucleotides are not properly complemented
 - [x] fixed primer positions (ranges) in root sequence when primer is reversed
